src/models/StyledNativeComponent.js

```diff
@@ -130,6 +130,8 @@ export default (constructWithOptions: Function, InlineStyle: Function) => {
 
       if (typeof innerRef === 'function') {
         innerRef(node)
+      } else if (typeof innerRef === 'object' && innerRef) {
```

Comment:
Hmm this might be too loose. Is there a ReactRef detector exposed by `react` or `react-is`?

Reply:
Sorry for the delay!

I've looked over the documented APIs, and didn't find anything. Digging deeper, they don't seem to have considered the case of *both* the component and the outer needing to ref the same element. They provide `React.forwardRef((props, ref) => node)`, but from testing it simply forwards the ref as-is - e.g. function to function, object to object, though invalid values are validated and rejected before the component is called.

I checked the react source code for refs, and [`createRef()` is nearly literally `return { current: null }`](https://github.com/facebook/react/blob/25dda90c1ecb0c662ab06e2c80c1ee31e0ae9d36/packages/react/src/ReactCreateRef.js#L12) and they seem to only bother with `typeof ref` - they are validated with `coerceRef(any) => null | FnRef | RefObject`](https://github.com/facebook/react/blob/25dda90c1ecb0c662ab06e2c80c1ee31e0ae9d36/packages/react-reconciler/src/ReactChildFiber.js#L101) and updated by [`commitAttachRef()`](https://github.com/facebook/react/blob/25dda90c1ecb0c662ab06e2c80c1ee31e0ae9d36/packages/react-reconciler/src/ReactFiberCommitWork.js#L321), which does have a dev mode warning with `ref.hasOwnProperty('current')`, but that seems like an implementation detail? I wouldn't want to assume that couldn't change at some point (e.g., made read-only by client code)

Seems like this "shared ref" use-case could be created as an issue, I didn't find an existing one.

Reply:
@gaearon could you weigh in on this? Wondering if `react-is` needs a new API to detect refs...

Reply:
Keep in mind any realistic changes to `createRef()` that could break the type check would probably also break the actual update `innerRef.current = node` below!

Reply:
The check looks fine to me. React only cares that it’s a plain object with `current` property. No plans to change this.

Reply:
Ok @simonbuchan can you do a simple check for if `ref.current` is defined then? If duck typing is sufficient, then adding that should square off this PR.

Reply:
@probablyup OK, added, though be aware react itself is perfectly happy to stamp `current` on any old `object` you give it. This change means your object will be silently ignored, I'd be happier with a warning similar to react's, but I don't know if there's a __DEV__ equivalent in styled-components?